harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -583,6 +610,42 @@ def make_workloads() -> workload.Stream:
         )
         controller.run()
 
+    # At the time of writing this test, supports_mixed_precision and
+    # supports_averaging_training_metrics are class methods, but we instantiate a
+    # PyTorchTrialController here in case that ever changes. We expect that a controller
+    # instantiated in the following way should always provide these supports.
+    def test_supports(self) -> None:
+        def make_workloads() -> workload.Stream:
+            trainer = utils.TrainAndValidate()
+            yield from trainer.send(steps=1, validation_freq=1, scheduling_unit=1)
+
+        controller = utils.make_trial_controller_from_trial_implementation(
+            trial_class=pytorch_xor_model.XORTrialAccessContext,
+            hparams=self.hparams,
+            workloads=make_workloads(),
+            trial_seed=self.trial_seed,
+        )
+        assert controller.supports_mixed_precision()
+        assert controller.supports_averaging_training_metrics()
+
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="no gpu available")
+    @pytest.mark.gpu
+    def test_apex_amp(self) -> None:
```

Comment:
after you convert this to the onevar model, I would like to see this test actually verify that the AMP is working, by feeding in an input that causes overflow or underflow and confirming that the scaling behavior is working.

Additionally, if you can confirm that the scaling behavior is properly preserved after loading from checkpoints, that would help us confirm/deny the existence of a long-standing bug report.

Reply:
I'm going to defer this development to a fresh branch.

Reply:
[DET-7885](https://determinedai.atlassian.net/browse/DET-7885)